I have not run them on my machine, on our Linux CI server I see:
```xml
      <test name="Templates.Test.GrpcTemplateTest.GrpcTemplateNativeAot" type="Templates.Test.GrpcTemplateTest" method="GrpcTemplateNativeAot" time="94.2551574" result="Fail">
        <output><![CDATA[| [0.001s] TestLifetime Information: Starting test GrpcTemplateNativeAot at 2023-05-01T23:40:29
| [0.002s] Templates.Test.GrpcTemplateTest Information: ==> /home/tester/aspnetcore/.dotnet/dotnet new grpc  --debug:disable-sdk-templates --debug:custom-hive "/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/Hives/b473dcd1-4727-479d-acb1-25d79f0e2d27/.templateengine" --aot -o /home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/]
| [0.948s] Templates.Test.GrpcTemplateTest Information: The template "ASP.NET Core gRPC Service" was created successfully.
| [0.949s] Templates.Test.GrpcTemplateTest Information: 
| [0.949s] Templates.Test.GrpcTemplateTest Information: Processing post-creation actions...
| [0.950s] Templates.Test.GrpcTemplateTest Information: Restoring /home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj:
| [1.996s] Templates.Test.GrpcTemplateTest Information:   Determining projects to restore...
| [56.579s] Templates.Test.GrpcTemplateTest Information:   Restored /home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj (in 53.68 sec).
| [56.624s] Templates.Test.GrpcTemplateTest Information: Restore succeeded.
| [56.624s] Templates.Test.GrpcTemplateTest Information: 
| [56.626s] Templates.Test.GrpcTemplateTest Information: 
| [56.627s] Templates.Test.GrpcTemplateTest Information: [ERROR] Failed to check update for Microsoft.DotNet.Web.ProjectTemplates.8.0::8.0.0-ci: Failed to check the update for the package Microsoft.DotNet.Web.ProjectTemplates.8.0::8.0.0-ci.
| [56.627s] Templates.Test.GrpcTemplateTest Information: [ERROR] Details: Value cannot be null. (Parameter 'releasePrefix')..
| [56.627s] Templates.Test.GrpcTemplateTest Information: [ERROR] 
| [56.649s] Templates.Test.GrpcTemplateTest Information: Process exited.
| [56.650s] Templates.Test.GrpcTemplateTest Information: Publishing ASP.NET Core application...
| [56.650s] Templates.Test.GrpcTemplateTest Information: ==> /home/tester/aspnetcore/.dotnet/dotnet publish  -c Release /bl  [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman]
| [56.879s] Templates.Test.GrpcTemplateTest Information: MSBuild version 17.7.0-preview-23225-01+6300d22b2 for .NET
| [58.085s] Templates.Test.GrpcTemplateTest Information:   Determining projects to restore...
| [93.857s] Templates.Test.GrpcTemplateTest Information:   Restored /home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj (in 34.97 sec).
| [93.996s] Templates.Test.GrpcTemplateTest Information: /home/tester/aspnetcore/.dotnet/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(287,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj]
| [94.150s] Templates.Test.GrpcTemplateTest Information: /home/tester/aspnetcore/.dotnet/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(478,5): error NETSDK1082: There was no runtime pack for Microsoft.AspNetCore.App available for the specified RuntimeIdentifier 'linux-x64'. [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj]
| [94.252s] Templates.Test.GrpcTemplateTest Information: Process exited.
| [94.257s] Templates.Test.GrpcTemplateTest Error: Test threw an exception.
| Xunit.Sdk.TrueException: Project new grpc  --aot failed to publish. Exit code 1.
| /home/tester/aspnetcore/.dotnet/dotnet publish  -c Release /bl \nStdErr: \nStdOut: MSBuild version 17.7.0-preview-23225-01+6300d22b2 for .NET
|   Determining projects to restore...
|   Restored /home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj (in 34.97 sec).
| /home/tester/aspnetcore/.dotnet/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(287,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj]
| /home/tester/aspnetcore/.dotnet/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(478,5): error NETSDK1082: There was no runtime pack for Microsoft.AspNetCore.App available for the specified RuntimeIdentifier 'linux-x64'. [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj]
| Expected: True
| Actual:   False
|    at Xunit.Assert.True(Nullable`1 condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 132
|    at Xunit.Assert.True(Boolean condition, String userMessage) in /_/src/xunit.assert/Asserts/BooleanAsserts.cs:line 116
|    at Templates.Test.Helpers.Project.RunDotNetPublishAsync(IDictionary`2 packageOptions, String additionalArgs, Boolean noRestore) in /_/src/ProjectTemplates/Shared/Project.cs:line 152
|    at Templates.Test.GrpcTemplateTest.GrpcTemplateCore(String[] args) in /_/src/ProjectTemplates/test/Templates.Tests/GrpcTemplateTest.cs:line 91
|    at Templates.Test.GrpcTemplateTest.GrpcTemplateNativeAot() in /_/src/ProjectTemplates/test/Templates.Tests/GrpcTemplateTest.cs:line 52
|    at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_0.<<InvokeTestMethodAsync>b__1>d.MoveNext() in /_/src/xunit.execution/Sdk/Frameworks/Runners/TestInvoker.cs:line 264
| --- End of stack trace from previous location ---
|    at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in /_/src/xunit.execution/Sdk/Frameworks/ExecutionTimer.cs:line 48
|    at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in /_/src/xunit.core/Sdk/ExceptionAggregator.cs:line 90
| [94.259s] TestLifetime Information: Finished test GrpcTemplateNativeAot in 94.2583206s
]]></output>
        <failure exception-type="Xunit.Sdk.TrueException">
          <message><![CDATA[Project new grpc  --aot failed to publish. Exit code 1.\n/home/tester/aspnetcore/.dotnet/dotnet publish  -c Release /bl \\nStdErr: \\nStdOut: MSBuild version 17.7.0-preview-23225-01+6300d22b2 for .NET\n  Determining projects to restore...\n  Restored /home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj (in 34.97 sec).\n/home/tester/aspnetcore/.dotnet/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(287,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj]\n/home/tester/aspnetcore/.dotnet/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(478,5): error NETSDK1082: There was no runtime pack for Microsoft.AspNetCore.App available for the specified RuntimeIdentifier 'linux-x64'. [/home/tester/aspnetcore/src/ProjectTemplates/test/Templates.Tests/bin/Release/net8.0/TestTemplates/AspNet.Pgpzatoryman/AspNet.Pgpzatoryman.csproj]\n\nExpected: True\nActual:   False]]></message>
          <stack-trace><![CDATA[   at Templates.Test.Helpers.Project.RunDotNetPublishAsync(IDictionary`2 packageOptions, String additionalArgs, Boolean noRestore) in /_/src/ProjectTemplates/Shared/Project.cs:line 152
   at Templates.Test.GrpcTemplateTest.GrpcTemplateCore(String[] args) in /_/src/ProjectTemplates/test/Templates.Tests/GrpcTemplateTest.cs:line 91
   at Templates.Test.GrpcTemplateTest.GrpcTemplateNativeAot() in /_/src/ProjectTemplates/test/Templates.Tests/GrpcTemplateTest.cs:line 52
--- End of stack trace from previous location ---]]></stack-trace>
        </failure>
      </test>
```

The CI job runs:
```
eng/build.sh --ci --arch x64 -c Release /p:IsPlaywrightAvailable=false --restore --build --pack
eng/build.sh --ci --arch x64 -c Release /p:IsPlaywrightAvailable=false --test
```

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -157,23 +157,9 @@
   <PropertyGroup>
     <BuildArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant())</BuildArchitecture>
 
-    <!-- This defines the list of RIDs supported by the ASP.NET Core shared framework. -->
-    <SupportedRuntimeIdentifiers>
-      win-x64;
-      win-x86;
-      win-arm;
-      win-arm64;
-      osx-x64;
-      osx-arm64;
-      linux-musl-x64;
-      linux-musl-arm;
-      linux-musl-arm64;
-      linux-x64;
-      linux-arm;
-      linux-arm64;
-      freebsd-x64
-    </SupportedRuntimeIdentifiers>
-
+    <!-- This defines the list of RIDs supported by the ASP.NET Core shared framework.
+         The list is a semicolon separated list of RIDs. Whitespace may not be added to the property. -->
+    <SupportedRuntimeIdentifiers>win-x64;win-x86;win-arm;win-arm64;osx-x64;osx-arm64;linux-musl-x64;linux-musl-arm;linux-musl-arm64;linux-x64;linux-arm;linux-arm64;freebsd-x64</SupportedRuntimeIdentifiers>
     <SupportedRuntimeIdentifiers Condition=" '$(PortableBuild)' == 'false' ">$(SupportedRuntimeIdentifiers);$(TargetRuntimeIdentifier)</SupportedRuntimeIdentifiers>
 
     <!-- Playwright provides binaries for Windows (x86 and x64), macOS (x64) and Linux (x64, non musl). We can't use it on other architectures. -->

--- a/eng/tools/GenerateFiles/GenerateFiles.csproj
+++ b/eng/tools/GenerateFiles/GenerateFiles.csproj
@@ -26,7 +26,7 @@
         MicrosoftNETCoreAppRuntimeVersion=$(MicrosoftNETCoreAppRuntimeVersion);
         MicrosoftPlaywrightCLIVersion=$(MicrosoftPlaywrightCLIVersion);
         LibNetHostAppPackVersion=$(BundledNETCoreAppPackageVersion);
-        SupportedRuntimeIdentifiers=$(SupportedRuntimeIdentifiers.Trim());
+        SupportedRuntimeIdentifiers=$([MSBuild]::Escape($(SupportedRuntimeIdentifiers)));
         ArtifactsShippingPackagesDir=$(ArtifactsShippingPackagesDir)
       </_TemplateProperties>
     </PropertyGroup>

